### PR TITLE
Add flip index ir nodes

### PR DESCRIPTION
**Pull request by jdconrad**
This change adds three nodes that support negative brace accesses into collections, arrays, and for def types. This helps separate that logic out from brace accesses in general. This also has some renaming for nodes adding an ir prefix to help differentiate ir node variables from user node variables.

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultUserTreeToIRTreePhase.java [hunk 1]
```diff
@@ -1368,20 +1372,31 @@ public void visitBrace(EBrace userBraceNode, ScriptScope scriptScope) {
...
+            irFlipArrayIndex.setIndexNode(injectCast(userBraceNode.getIndexNode(), scriptScope));
+
             BraceSubNode irBraceSubNode = new BraceSubNode();
-            irBraceSubNode.setChildNode(injectCast(userBraceNode.getIndexNode(), scriptScope));
+            irBraceSubNode.setIndexNode(irFlipArrayIndex);
             irBraceSubNode.setLocation(userBraceNode.getLocation());
             irBraceSubNode.setExpressionType(scriptScope.getDecoration(userBraceNode, ValueType.class).getValueType());
             irExpressionNode = irBraceSubNode;
         } else if (prefixValueType == def.class) {
+            ;
```
Comment (stu-elastic) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultUserTreeToIRTreePhase.java:1386]: This semi is lonely.

**Review by stu-elastic — APPROVED** [review]

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultUserTreeToIRTreePhase.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultUserTreeToIRTreePhase.java]: Good catch. Fixed!

**Comment by jdconrad** [comment]
@stu-elastic Thanks for the review! Will commit as soon as CI passes.

**Comment by jdconrad** [comment]
@elasticsearchmachine run elasticsearch-ci/packaging-sample-windows

**Pull request merged**